packages/api/src/Api.ts

```diff
@@ -33,18 +33,43 @@ const Types = require('@cennznet/types');
 
 export class Api extends ApiPromise {
     static async create(options: ApiOptions | ProviderInterface = {}): Promise<Api> {
-        return new Api(options).isReady;
+        return new Api(options).isReady as Promise<Api>;
     }
 
     // TODO: add other crml namespaces
+    
+    private _genericAsset?: GenericAsset;
+    private _cennzxSpot?: CennzxSpot;
+
     /**
      * Generic Asset CRML extention
      */
-    genericAsset?: GenericAsset;
+    get genericAsset(): GenericAsset {
```

Comment:
we use `Object.defineProperty` to set genericAsset and cennzxSpot, so i think you can just throw Error in the getter, (without creating and visiting `_genericAsset`)

Reply:
That sounds like a fair optimisation to me!